game.js

```diff
@@ -0,0 +1,148 @@
+let bot = require("./bot.js");
+let helpers = require("./helpers.js");
+let lastLetter = "";
+let cities = [
+  "Архангельск",
+  "Борисполь",
+  "Воронеж",
+  "Гюмри",
+  "Донецк",
+  "Енакиево",
+  "Жуковский",
+  "Запорожье",
+  "Изюм",
+  "Йошкар-Ола",
+  "Киев",
+  "Лондон",
+  "Москва",
+  "Николаев",
+  "Орёл",
+  "Париж",
+  "Ростов",
+  "Селидово",
+  "Трускавец",
+  "Уфа",
+  "Флоренция",
+  "Харьков",
+  "Цюрих",
+  "Челябинск",
+  "Шанхай",
+  "Щёлково",
+  "Ыгдыр",
+  "Электросталь",
+  "Юрмала",
+  "Ялта"
+].map(item => item.toLowerCase());
+
+exports.session = class Session {
+  constructor(id, chatId = 0) {
+    this.id = id;
+    this.chatId = chatId;
+    this.spentCities = new Set();
+  }
+};
+
+function randomCity(arrCities) {
+  return arrCities[helpers.getRandomInt(arrCities.length)];
+}
+
+function lastValidLetter(str) {
+  let lastLetter = str[str.length - 1];
+  let invalidLetters = ["ь", "Ъ"];
+  let i = 2;
+
+  while (invalidLetters.includes(lastLetter) || str.length == i) {
+    lastLetter = str[str.length - i];
+    i++;
+  }
+  return lastLetter;
+}
+
+function checkCityInDB(sessionId, city, letter) {
+  let result = { isError: false, errorMsg: "" };
```

Comment:
Do not mutate variable. Just return an error message or `null`. `string | null`

Reply:
fixed it

Reply:
ready

Reply:
```suggestion
function checkCityInDB(chatID, city, letter) {
  if (letter != city[0]) {
    return "Нужно назвать город на букву " + letter.toUpperCase();
  }
  if (sessions[chatID].spentCities.has(city)) {
    return "Этот город уже был назван!";
  }
  return null;
}
```